packages/dataviews/src/components/dataviews-filters/filter.tsx

```diff
@@ -494,10 +494,24 @@ export default function Filter( {
 			return filterInView?.value?.includes( element.value );
 		} );
 	} else if ( filterInView?.value !== undefined ) {
+		// Find the field to check its type
```

Comment:
This is a proof of concept. Consider using `getValue` from the field (see note [below](https://github.com/WordPress/gutenberg/pull/72756/files#r2470781450)) instead. Note the filter's input-widget prepares [the field in a specific way](https://github.com/WordPress/gutenberg/blob/5036924c01c7ad430672cc6345a89962ceb8fdc6/packages/dataviews/src/components/dataviews-filters/input-widget.tsx#L58) (because it doesn't have access to any item).